Source/player.cpp

```diff
@@ -364,16 +366,20 @@ void StartWalk(int pnum, Displacement vel, Direction dir, bool pmWillBeCalled)
 	StartWalkAnimation(player, dir, pmWillBeCalled);
 }
 
-void SetPlayerGPtrs(const char *path, std::unique_ptr<byte[]> &data, std::array<std::optional<CelSprite>, 8> &anim, int width)
+void SetPlayerGPtrs(Player &player, const char *path, std::unique_ptr<byte[]> &data, std::array<std::optional<CelSprite>, 8> &anim, int width)
 {
 	data = nullptr;
 	data = LoadFileInMem(path);
 	if (data == nullptr && gbQuietMode)
 		return;
 
-	const byte *directionFrames[8];
+	byte *directionFrames[8];
 	CelGetDirectionFrames(data.get(), directionFrames);
+	auto trn = GetClassTRN(player);
 	for (size_t i = 0; i < 8; i++) {
+		if (trn != nullptr) {
+			Cl2ApplyTrans(directionFrames[i], reinterpret_cast<std::array<uint8_t, 256> &>(trn[0]), static_cast<int>(directionFrames[i][0]));

```

Comment:
Previously we didn't have a `Cl2ApplyTrans` here for warrior/rogue/etc.
Why apply these now? For modding?

Reply:
Nevermind, I've now read the PR description :smile: